src/main/java/com/google/cloud/discotoproto3converter/disco/Method.java

```diff
@@ -65,11 +65,16 @@ public static Method from(DiscoveryNode root, Node parent) {
         pathParams.put(name, schema);
       } else if (schema.location().toLowerCase().equals("query")) {
         queryParams.put(name, schema);
-      } if (schema.required()) {
-        requiredParams.put(name, schema);
       }
     }
 
+    List<String> requiredParamNames =
+        root.getArray("parameterOrder")
```

Comment:
For the discovery doc schema spec:

_`methods.(key).parameterOrder[]` - `list` 
Ordered list of **required** parameters. This serves as a hint to **clients on how to structure their method signatures**. The array is ordered such that the most significant parameter appears first._

I believe we are using the `parameterOrder` property literally for what it exists: to specify client methods signature. The check for required is not need (I actually had it originally but then removed), because the fields in the list are guaranteed to be requried.

Reply:
Sweet!